pkg/ebpf/alu_instructions.go

```diff
@@ -191,6 +230,12 @@ func Xor[T Src](dstReg pb.Reg, src T) *pb.Instruction {
 // Mov64 Creates a new 64 bit Mov instruction that is either imm or reg depending
 // on the data type of src
 func Mov64[T Src](dstReg pb.Reg, src T) *pb.Instruction {
+	// The ALU instruction requires explicit indication of int64 if the
+	// instruction has a 64-bit immediate value. Otherwise, it will interpret
+	// the value as an int
+	if int(int32(src)) != int(int64(src)) {
```

Comment:
Could we add a test case where we make sure that int32 -> emits an alu mov, int64 -> emits a load instruction?

Reply:
I think there is already a test case for it:
https://github.com/google/buzzer/blob/edab5305d4ca6af0dcbb3b935c3cd39151c84b6d/pkg/ebpf/alu_instructions_test.go#L308C4-L308C73